control/veda_deliver_cielo.py

```diff
@@ -1,186 +1,228 @@
-
-        r5 = requests.get(
-            ''.join((
-                self.c24_site,
-                self.transcribe,
-                '?v=1&api_token=',
-                api_token,
-                '&job_id=',
-                job_id,
-                '&transcription_fidelity=',
-                self.c24_defaults['c24_fidelity'],
-                '&priority=',
-                self.c24_defaults['c24_speed']
-            ))
-        )
-        return ast.literal_eval(r5.text)['TaskId']
 
-    def retrieve_defaults(self):
-        video_query = Video.objects.filter(
-            edx_id=self.veda_id
-        ).latest()
+class Cielo24AddMediaError(Cielo24Error):
+    """
+    An error occurred during add media.
+    """
+    pass
+
 
-        if video_query.inst_class.mobile_override is True:
+class Cielo24PerformTranscriptError(Cielo24Error):
+    """
+    An error occurred during perform transcript.
+    """
+    pass
+
+
+class Cielo24Transcript(object):
+    """
+    Cielo24 Integration
+    """
+    def __init__(self, video, video_transcript_preferences):
+        self.video = video
+        self.video_transcript_preferences = video_transcript_preferences
+        # Defaults
+        self.cielo24_site = 'https://api.cielo24.com/api'
+        self.cielo24_new_job = '/job/new'
+        self.cielo24_add_media = '/job/add_media'
+        self.cielo24_perform_transcription = '/job/perform_transcription'
+        self.callback_base_url = reverse('cielo24_transcript_completed')
+
+        # Set Cielo24 defaults
+        self.cielo24_defaults = self.set_defaults()
+
+    def set_defaults(self):
+        """
+        Set default values for cielo24.
+        """
+        if self.video.inst_class.mobile_override is True:
             url_query = URL.objects.filter(
-                videoID=video_query,
+                videoID=self.video,
                 encode_url__icontains='_LBO.mp4',
             ).latest()
         else:
             url_query = URL.objects.filter(
-                videoID=video_query,
+                videoID=self.video,
                 encode_url__icontains='_DTH.mp4',
             ).latest()
-        if video_query.inst_class.c24_username is None:
-            ErrorObject.print_error(
-                message='Cielo24 Record Incomplete',
-            )
-            return None
-
-        c24_defaults = {
-            'c24_user': video_query.inst_class.c24_username,
-            'c24_pass': video_query.inst_class.c24_password,
-            'c24_speed': video_query.inst_class.c24_speed,
-            'c24_fidelity': video_query.inst_class.c24_fidelity,
-            'edx_id': self.veda_id,
+
+        cielo24_defaults = {
+            'api_key': self.video_transcript_preferences['api_key'],
+            'turnaround': self.video_transcript_preferences['cielo24_turnaround'],
+            'fidelity': self.video_transcript_preferences['cielo24_fidelity'],
+            'edx_id': self.video.edx_id,
             'url': url_query.encode_url
```

Comment:
we may want to use bucket url instead of cloudfront OR whitelist cielo24 and 3playmedia.